### DISCUSSION TITLE
readByQuery limit default

### DISCUSSION THREAD

**Discussion by keesvanbemmel** [directus/directus#15455] [Archive]
Hi all!

I was wondering why there is a default limit of 100 for the readByQuery items service function. It seems to me that, since we're filtering anyway, the limit is moot and should be set to -1 by default.

Would love to hear your feedback!

Cheers,

Kees.

**Comment by rijkvanzanten** [comment]
It's mostly that having it unlimited by default can be an easy footgun on large data sets. If you have a table of 15 million rows, defaulting to `-1` will immediately crash your whole request if you forget to add the limit. It also ensures that any internal code—even though it might cause unexpected bugs—will never read a whole table into memory by default
